### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by Bondok6**
# Hello there 🙋‍♂️,

### Thanks for taking the time to review our project.
<hr>

## What was done in this PR by @eduardosancho & @Bondok6 🧾

- Install CanCanCan in our project.
- Add a role column to the `users` table. 
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin").
  - For that, we implemented the post deleting functionality. Add the `Delete` button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin")
  - For that, we implemented the comment deleting functionality. Add the `Delete` button to the view and make sure that only authorized users can see it.

Gif found on CodeReviewGifs.com hosted on Giphy.com

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -38,4 +36,12 @@ def create
       end
     end
   end
+
+  def destroy
+    post = Post.find(params[:id])
+    post.destroy
+
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [x] you should Use CanCanCan for Post authorization in Post Controller like below:
```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end

```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (eduardosancho) [reply app/controllers/posts_controller.rb]: Hi @mikethreels, thanks for the review, we would some insight on a problem caused by `load_and_authorize_resource`.

## When we try to create a post, like this:
image

## We get the following error:
image
image

## This is our PostsControllers#create action:
image

Comment (eduardosancho) [reply app/controllers/posts_controller.rb]: Hi, we found a [solution](https://github.com/eduardosancho/blog-app/pull/11/commits/5cf0a98160edc2067c52acb5b821cdd509951503), we moved the permitted parameters for the #create action into a private hash.

**Review by rukundoeric — APPROVED** [review]
Hi @eduardosancho,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**